2006-02-22  Arnaud Charlet  <charlet@adacore.com>

	* src/cairo-win32-surface.c: Fix build failure under Win32
	Submitted at https://bugs.freedesktop.org/show_bug.cgi?id=6015

	* pixman/src/pixman.h: Fix build failures on ia64/pa hpux
	Submitted at https://bugs.freedesktop.org/show_bug.cgi?id=6016

--- src/cairo-win32-surface.c.old	2005-10-03 21:44:43.000000000 +0100
+++ src/cairo-win32-surface.c	2006-02-20 14:40:30.000000000 +0100
@@ -1062,6 +1062,7 @@
 CRITICAL_SECTION cairo_toy_font_face_hash_table_mutex;
 CRITICAL_SECTION cairo_scaled_font_map_mutex;
 CRITICAL_SECTION cairo_ft_unscaled_font_map_mutex;
+CRITICAL_SECTION _global_image_glyph_cache_mutex;
 
 BOOL WINAPI
 DllMain (HINSTANCE hinstDLL,
@@ -1075,11 +1076,13 @@
     InitializeCriticalSection (&cairo_toy_font_face_hash_table_mutex);
     InitializeCriticalSection (&cairo_scaled_font_map_mutex);
     InitializeCriticalSection (&cairo_ft_unscaled_font_map_mutex);
+    InitializeCriticalSection (&_global_image_glyph_cache_mutex);
     break;
   case DLL_PROCESS_DETACH:
     DeleteCriticalSection (&cairo_toy_font_face_hash_table_mutex);
     DeleteCriticalSection (&cairo_scaled_font_map_mutex);
     DeleteCriticalSection (&cairo_ft_unscaled_font_map_mutex);
+    DeleteCriticalSection (&_global_image_glyph_cache_mutex);
     break;
   }
   return TRUE;
--- pixman/src/pixman.h.orig	Mon Feb 27 11:08:33 2006
+++ pixman/src/pixman.h	Mon Feb 27 11:08:11 2006
@@ -94,6 +94,8 @@ SOFTWARE.
   typedef unsigned __int32 uint32_t;
   typedef __int64 int64_t;
   typedef unsigned __int64 uint64_t;
+#elif defined (__hpux__)
+#include <sys/types.h>
 #else
 # include <stdint.h>
 #endif
